File: Problemset/lian-xu-zi-shu-zu-de-zui-da-he-lcof/lian-xu-zi-shu-zu-de-zui-da-he-lcof.py

# @Title: 连续子数组的最大和 (连续子数组的最大和  LCOF)
# @Author: cocofe
# @Date: 2020-12-11 22:22:29
# @Runtime: 52 ms
# @Memory: 15 MB

class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = float('-inf')
        prev = 0
        for num in nums:
            curr = max(num, num+prev)
            ans = max(ans, curr)
            prev = curr
        return ans

File: Problemset/lian-xu-zi-shu-zu-de-zui-da-he-lcof/lian-xu-zi-shu-zu-de-zui-da-he-lcof.java

// @Title: 连续子数组的最大和 (连续子数组的最大和  LCOF)
// @Author: cocofe
// @Date: 2021-07-17 00:44:55
// @Runtime: 1 ms
// @Memory: 45 MB

class Solution {
    public int maxSubArray(int[] nums) {
        int sum = 0;
        int ans = Integer.MIN_VALUE;
        for(int num : nums) {
            if (sum + num <= 0) {
                ans = Math.max(Math.max(sum+num, num), ans);
                sum = 0;
            } else {
                sum += num;
                ans = Math.max(sum, ans);
            }
        }
        return ans;

    }
}
